### PR TITLE
Sidebar design & behavior improvements

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

Apply design improvements on the Sidebar component based on VTEX client's feedback

List of improvements:

* Apply spacing changes over the section and section items
* Change the sidebar behavior to show a preview of each section items when hovering over the section's icon

#### How should this be manually tested?

1. Check that the spacing of the section items changes between the old version and the new one
2. Hover the icons using your mouse to check the section preview
3. Hover with your mouse on the last section and see that the URL preview doesn't override the icon

#### Screenshots or example usage

> 💡 You can type "a", "s", and "t" to hide storybook gadgets.

**Before**

You can check this [storybook workspace](https://leftnav-before.surge.sh/?path=/story/admin-ui-sidebar--playground) for a preview without the changes applied.

**After**

You can check the workspace preview on the storybook URL generated by our CI

leftnav-hover

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

**Comment by marcelovicentegc** [comment]
No more storybook preview?

**Comment by marcelovicentegc** [comment]
We could add a delay while applying the hover style back to the selected item (or even apply a different style). Notice how the selected item's style "glitches" while hovering over different sections:

https://user-images.githubusercontent.com/33183880/157429837-34624313-2c7c-4184-932f-69b45f81a2a3.mov

**Comment by lucasaarcoverde** [comment]
> No more storybook preview automatically generated?

We are not deploying it on V ercel anymore, so you can find the generated preview on the PR checks

Captura de Tela 2022-03-09 às 11 35 11

**Review by Lenderson — APPROVED** [review]

File: packages/admin-ui/src/components/Sidebar/__stories__/Sidebar.stories.tsx [hunk 1]
```diff
@@ -147,29 +154,41 @@ export const Playground: Story<any> = (args) => {
...
-          alignItems: 'center',
-          border: '4px dashed black',
-          margin: 1,
-        }}
-      >
-        TOP BAR
-      </Box>
+      <Topbar>
+        <TopbarStart>
+          <Set spacing={3}>
```
Comment (anitavincent) [inline packages/admin-ui/src/components/Sidebar/__stories__/Sidebar.stories.tsx:165]: Do spacing tokens work here?

File: packages/admin-ui/src/components/Sidebar/__stories__/Sidebar.stories.tsx [hunk 2]
```diff
@@ -182,6 +201,9 @@ export const Playground: Story<any> = (args) => {
                 icon={item.icon}
                 key={item.label}
                 selected={index === 0}
+                onMouseEnter={() => {
```
Comment (anitavincent) [inline packages/admin-ui/src/components/Sidebar/__stories__/Sidebar.stories.tsx:204]: hi!

**Review by anitavincent — APPROVED** [review]

File: packages/admin-ui/src/components/Sidebar/__stories__/Sidebar.stories.tsx (hunk 1 above)
Comment (lucasaarcoverde) [reply packages/admin-ui/src/components/Sidebar/__stories__/Sidebar.stories.tsx:165]: Not yet, we should fix it once we review this component

**Pull request merged**